### PR TITLE
Connect hypervisor and visor over dmsg

### LINKED ISSUE #78: Implement Hypervisor <--> Visor connection over dmsg
**Feature description**

This is the old PR pointing to the old repo: https://github.com/skycoin/skywire/pull/563/files

### PR DESCRIPTION
**Pull request by Kifen**
Fixes #78 	

 Changes:	
-	Copied this [PR](https://github.com/SkycoinProject/skywire-mainnet/pull/79)

How to test this PR:
Run an integration env, view connected visors in the hypervisor window.

### CODE REVIEW

File: pkg/hypervisor/config.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ import (
 	"path/filepath"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/internal/skyenv"
```
Comment (nkryuchkov) [inline pkg/hypervisor/config.go]: Could you please move lines 11 and 12 below and group with line 16?

File: pkg/hypervisor/hypervisor.go [hunk 2]
```diff
@@ -6,15 +6,16 @@ import (
...
-	"net"
 	"net/http"
 	"net/rpc"
 	"strconv"
+	"strings"
 	"sync"
 	"time"
 
+	"github.com/SkycoinProject/dmsg"
+
```
Comment (nkryuchkov) [inline pkg/hypervisor/hypervisor.go]: We can remove this empty line

File: pkg/visor/rpc_client.go [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+	"errors"
 	"fmt"
 	"math/rand"
+	"net"
 	"net/http"
 	"net/rpc"
 	"sync"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet"
```
Comment (nkryuchkov) [inline pkg/visor/rpc_client.go]: Could you please group imports?

File: pkg/visor/visor.go [hunk 4]
```diff
@@ -20,9 +20,10 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/SkycoinProject/skywire-mainnet/pkg/httputil"
```
Comment (nkryuchkov) [inline pkg/visor/visor.go]: Could you please group imports?

**Review by nkryuchkov — COMMENTED** [review]
Great job!

**Review by Darkren — APPROVED** [review]
Other than the above mentioned comments, great job!

File: pkg/hypervisor/config.go (hunk 1 above)
Comment (Kifen) [reply pkg/hypervisor/config.go]: done

File: pkg/hypervisor/hypervisor.go (hunk 2 above)
Comment (Kifen) [reply pkg/hypervisor/hypervisor.go]: Done

File: pkg/visor/rpc_client.go (hunk 3 above)
Comment (Kifen) [reply pkg/visor/rpc_client.go]: done

File: pkg/visor/visor.go (hunk 4 above)
Comment (Kifen) [reply pkg/visor/visor.go]: Done

File: pkg/router/routerclient/client_test.go [hunk 5]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/SkycoinProject/skywire-mainnet/internal/testhelpers"
 
 	"github.com/SkycoinProject/dmsg/cipher"
+
```
Comment (nkryuchkov) [inline pkg/router/routerclient/client_test.go]: Could you please split and sort the imports into three groups? Standard library, packages other than `skywire-mainnet` and `skywire-mainnet`? Thanks!

**Review by nkryuchkov — CHANGES_REQUESTED** [review]
Just a small change 🙂

File: pkg/router/routerclient/client_test.go (hunk 5 above)
Comment (Kifen) [reply pkg/router/routerclient/client_test.go]: Done

**Review by nkryuchkov — APPROVED** [review]

**Pull request merged**